Add doxygen for the assigned functions: H5Pregister1,H5Pinsert1,H5Pen…

kyang2014 opened the pull request:
…code1, H5Pget_filter_by_id1,H5Pget_version, H5Pset_file_space,H5Pget_file_space. Someone already adds H5Pget_filter1. Also fixs an extra parameter 'close' call back function for HPregister2.

kyang2014 committed:
Add doxygen for the assigned functions: H5Pregister1,H5Pinsert1,H5Pencode1, H5Pget_filter_by_id1,H5Pget_version, H5Pset_file_space,H5Pget_file_space. Someone already adds H5Pget_filter1. Also fixs an extra parameter 'close' call back function for HPregister2.

kyang2014 committed:
doxygen work. fixs format by using clang-format.

kyang2014 commented:
Cannot choose the reviewers and see if I close and reopen it can do the trick.

gheber commented on src/H5Ppublic.h:
```diff
@@ -6911,16 +6907,545 @@ H5_DLL herr_t H5Pset_mcdt_search_cb(hid_t plist_id, H5O_mcdt_search_cb_t func, v
+ * given the opportunity to close any temporary data structures that were
+ * set up when the routine was called. The C++ application should save some
+ * state as the routine is started so that any problem that occurs might be
+ * diagnosed.
```
Let's not repeat this block literally everywhere! Would you mind appending the following single line to `aliases`?
```
ALIASES += cpp_c_api_note="\note \Bold{C++ Developers using HDF5 C-API functions beware:}\nIf a C routine that takes a function pointer as an argument is called from within C++ code, the C routine should be returned from normally. Examples of this kind of routine include callbacks such as H5Pset_elink_cb() and H5Pset_type_conv_cb() and functions such as H5Tconvert() and H5Ewalk2().\n Exiting the routine in its normal fashion allows the HDF5 C library to clean up its work properly. In other words, if the C++ application jumps out of the routine back to the C++ \c catch statement, the library is not given the opportunity to close any temporary data structures that were set up when the routine was called. The C++ application should save some state as the routine is started so that any problem that occurs might be diagnosed."
```
Then you can replace this paragraph by just `\cpp_c_api_note`

gheber commented on src/H5Ppublic.h:
```diff
@@ -6911,16 +6907,545 @@ H5_DLL herr_t H5Pset_mcdt_search_cb(hid_t plist_id, H5O_mcdt_search_cb_t func, v
+ * given the opportunity to close any temporary data structures that were
+ * set up when the routine was called. The C++ application should save some
+ * state as the routine is started so that any problem that occurs might be
+ * diagnosed.
```
See the comment above.

gheber commented on src/H5Ppublic.h:
```diff
@@ -6911,16 +6907,545 @@ H5_DLL herr_t H5Pset_mcdt_search_cb(hid_t plist_id, H5O_mcdt_search_cb_t func, v
+ *          this property is being created. The #H5P_prp_create_func_t
+ *          callback function is defined as follows:
+ *
+ *          \snippet this H5P_prp_cb1_t_snip
```
Using snippets for callbacks is the way to go. Great! But we shouldn't repeat the parameter lists everywhere. Since Doxygen hyperlinks everything, the user has to just click on the callback type and will be taken to the documentation. The parameters should be defined once with the callback. As an example see `H5E_walk2_t` in `H5Epublic.h` around line 159. Does that make sense?

gheber commented on src/H5Ppublic.h:
```diff
@@ -6911,16 +6907,545 @@ H5_DLL herr_t H5Pset_mcdt_search_cb(hid_t plist_id, H5O_mcdt_search_cb_t func, v
+ *          the property. The #H5P_prp_set_func_t callback function is defined
+ *          as follows:
+ *
+ *          \snippet this H5P_prp_cb2_t_snip
```
Ditto. Let's not repeat parameter lists of callbacks.

gheber commented on src/H5Ppublic.h:
```diff
@@ -6911,16 +6907,545 @@ H5_DLL herr_t H5Pset_mcdt_search_cb(hid_t plist_id, H5O_mcdt_search_cb_t func, v
+ *          property value. The #H5P_prp_get_func_t callback function is
+ *          defined as follows:
+ *
+ *          \snippet this H5P_prp_cb2_t_snip
```
Ditto. Let's not repeat parameter lists of callbacks.

gheber commented on src/H5Ppublic.h:
```diff
@@ -6911,16 +6907,545 @@ H5_DLL herr_t H5Pset_mcdt_search_cb(hid_t plist_id, H5O_mcdt_search_cb_t func, v
+ *          deleted from a property list. The #H5P_prp_delete_func_t
+ *          callback function is defined as follows:
+ *
+ *          \snippet this H5P_prp_cb2_t_snip
```
Ditto. Let's not repeat parameter lists of callbacks.

gheber commented on src/H5Ppublic.h:
```diff
@@ -6911,16 +6907,545 @@ H5_DLL herr_t H5Pset_mcdt_search_cb(hid_t plist_id, H5O_mcdt_search_cb_t func, v
+ *          this property is being created through a \p prp_copy operation.
+ *          The #H5P_prp_copy_func_t callback function is defined as follows:
+ *
+ *          \snippet this H5P_prp_cb1_t_snip
```
Ditto. Let's not repeat parameter lists of callbacks.

gheber commented on src/H5Ppublic.h:
```diff
@@ -6911,16 +6907,545 @@ H5_DLL herr_t H5Pset_mcdt_search_cb(hid_t plist_id, H5O_mcdt_search_cb_t func, v
+ *          property is being closed. The #H5P_prp_close_func_t callback
+ *          function is defined as follows:
+ *
+ *          \snippet this H5P_prp_cb1_t_snip
```
Ditto. Let's not repeat parameter lists of callbacks.

gheber commented on src/H5Ppublic.h:
```diff
@@ -6911,16 +6907,545 @@ H5_DLL herr_t H5Pset_mcdt_search_cb(hid_t plist_id, H5O_mcdt_search_cb_t func, v
+ *          The \p prp_set routine is called before a new value is copied
+ *          into the property. The #H5P_prp_set_func_t callback function
+ *          is defined as follows:
+ *          \snippet this H5P_prp_cb2_t_snip
```
Ditto. Let's not repeat parameter lists of callbacks.

gheber commented on src/H5Ppublic.h:
```diff
@@ -6911,16 +6907,545 @@ H5_DLL herr_t H5Pset_mcdt_search_cb(hid_t plist_id, H5O_mcdt_search_cb_t func, v
+ *          a property value. The #H5P_prp_get_func_t callback function
+ *          is defined as follows:
+ *
+ *          \snippet this H5P_prp_cb2_t_snip
```
Ditto. Let's not repeat parameter lists of callbacks.

gheber commented on src/H5Ppublic.h:
```diff
@@ -6911,16 +6907,545 @@ H5_DLL herr_t H5Pset_mcdt_search_cb(hid_t plist_id, H5O_mcdt_search_cb_t func, v
+ *          deleted from a property list. The #H5P_prp_delete_func_t
+ *          callback function is defined as follows:
+ *
+ *          \snippet this H5P_prp_cb2_t_snip
```
Ditto. Let's not repeat parameter lists of callbacks.

gheber commented on src/H5Ppublic.h:
```diff
@@ -6911,16 +6907,545 @@ H5_DLL herr_t H5Pset_mcdt_search_cb(hid_t plist_id, H5O_mcdt_search_cb_t func, v
+ *
+ *          The #H5P_prp_copy_func_t callback function is defined as follows:
+ *
+ *          \snippet this H5P_prp_cb1_t_snip
```
Ditto. Let's not repeat parameter lists of callbacks.

gheber commented on src/H5Ppublic.h:
```diff
@@ -6911,16 +6907,545 @@ H5_DLL herr_t H5Pset_mcdt_search_cb(hid_t plist_id, H5O_mcdt_search_cb_t func, v
+ *          property is being closed.
+ *
+ *          The #H5P_prp_close_func_t callback function is defined as follows:
+ *          \snippet this H5P_prp_cb1_t_snip
```
Ditto. Let's not repeat parameter lists of callbacks.

gheber reviewed (approved):
Great job. Let's eliminate a few repetitions if you have time! See the comments inline. Every repetition is the potential for inconsistency and we just had too much of that.

qkoziol commented on src/H5Ppublic.h:
```diff
@@ -1582,16 +1582,12 @@ H5_DLL int H5Piterate(hid_t id, int *idx, H5P_iterate_t iter_func, void *iter_da
  *          function is defined as follows:
  *
- *          \snippet this H5P_prp_cb2_t_snip
+ *          \snippet this H5P_prp_cb1_t_snip
```
The 'cb1' / 'cb2' types are just convenience / maintenance helpers in the header file - the documentation should use the types for the parameters in the H5P* routines.

qkoziol replied:
e.g. H5P_prp_create_func_t, H5P_prp_set_func_t, etc.

kyang2014 committed:
doxgen work for H5Pregister1 etc. Addressed Barbara and Gerd's comments.
For Quincey's comments, since we are not supposed to change the source code.
I leave this to future improvements.
